Gems/ROS2/Code/Include/ROS2/Sensor/Events/SensorEventSource.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+namespace ROS2
+{
+    struct SensorConfiguration;
+
+    //! Base class template for sensor event sources (based e.g. on TickBus or engine physics callback). Developer that wants to implement
+    //! new event source, should derive it from this class and signal m_sensorEvent based on chosen approach. Derived classes should rather
+    //! not auto connect any handler to m_sensorEvent, unless it is done to manage their internal logic. This event is for the final user
+    //! that can connect prepared event handler to it using ConnectToSourceEvent method.
+    //! @see ROS2::TickBasedSource.
+    //! @see ROS2::PhysicsBasedSource.
+    template<template<typename...> class EventT, template<typename...> class EventHandlerT, typename... EventArgs>
+    class SensorEventSource
+    {
+    public:
+        using SourceBaseType = SensorEventSource<EventT, EventHandlerT, EventArgs...>;
+        using SourceEventType = EventT<EventArgs...>;
+        using SourceEventHandlerType = EventHandlerT<EventArgs...>;
+
+        virtual ~SensorEventSource() = default;
+
+        //! Sets up event source - ee event source description for more details. After call to this method event source is supposed to start
+        //! signalling source event.
+        virtual void Activate()
+        {
+        }
+
+        //! Shuts down event source - see event source description for more details. After call to this method event source is supposed to
+        //! stop signalling source event.
+        virtual void Deactivate()
+        {
+        }
+
+        //! Configures event source based on sensor settings. Details depend on specific event source implementation.
+        //! @param sensorConfiguration Implemented sensor configuration.
+        //! @note In most cases it will be safer to call this before Activate, however at the end it depends on specific event source
+        //! implementations.
+        virtual void Configure([[maybe_unused]] const SensorConfiguration& sensorConfiguration)
```

Comment:
Not entirely sure about purpose of this member function. It seems fields other than frequency are not related.